### ISSUE TITLE
Implement Sinkhorn normalization

### ISSUE THREAD

**Issue by mateusznowakTT** [tenstorrent/tt-metal#43233]
Implement Sinkhorn normalization operation using generic_op framework as a single kernel (not composite)

Targeting tensors of shape [B, n_tokens, 4, 4], starting with B=n_tokens=1, and then expanding to larger values on those dimensions.

Torch reference implementation is as follows:
```
import torch
import torch.nn.functional as F

def sinkhorn_normalize(x: torch.Tensor, iters: int = 20, eps: float = 1e-6) -> torch.Tensor:
    """
    Sinkhorn normalization for MHC mixing matrices.
    
    Args:
        x: Input tensor of shape [num_tokens, hc_mult, hc_mult]
        iters: Number of row/column normalization iterations
        eps: Small constant for numerical stability
    
    Returns:
        Doubly-stochastic matrices of same shape
    """
    # Initial softmax over last dimension (rows sum to 1)
    x = F.softmax(x, dim=-1) + eps
    
    # First column normalization
    x = x / (x.sum(dim=-2, keepdim=True) + eps)
    
    # Alternate row/column normalization
    for _ in range(iters - 1):
        x = x / (x.sum(dim=-1, keepdim=True) + eps)  # row norm
        x = x / (x.sum(dim=-2, keepdim=True) + eps)  # col norm
```

**Comment by minaliuTT** [comment]
@cmaryanTT Is this a P0 or P1?

**Comment by cmaryanTT** [comment]
This is materially done and it belongs in the Blaze repo. @mateusznowakTT can close this unless he’s using it to track something else.

**Comment by mateusznowakTT** [comment]
Yes, no longer planning to use it here, closing.
